app/models/spotlight/page.rb

```diff
@@ -93,5 +101,26 @@ def should_display_title?
     def lock!(user)
       create_lock(by: user).tap(&:current_session!) unless lock.present?
     end
+
+    def updated_after?(other_page)
+      return false unless other_page
+      updated_at > other_page.updated_at
+    end
+
+    def translated_page_for(locale)
+      translated_pages.for_locale(locale).first
+    end
+
+    def clone_for_locale(locale)
+      dup.tap do |np|
+        np.locale = locale
+        np.default_locale_page = self
+        np.published = false
+
+        if !top_level_page? && (parent_translation = parent_page.translated_page_for(locale)).present?
+          np.parent_page = parent_translation
+        end
```

Comment:
NOTE: This requires that the parent translated page be created before the child page.  We could also do the inverse where we check when a top level page is created if there have been any child pages for the given language created that are associated with the default language page (meaning that the child page had been created before the parent page).  Not sure what folks think about this or if this is just the wrong approach in general.